components/feature/media/src/main/java/mozilla/components/feature/media/service/MediaSessionServiceDelegate.kt

```diff
@@ -116,7 +127,16 @@ internal class MediaSessionServiceDelegate(
         }
 
         updateMediaSession(state)
-        updateNotification(state)
+        CoroutineScope(notificationDispatcher).launch() {
+            updateNotification(state)
+        }
```

Comment:
I wonder if this needs to be more complex: Can loading the artwork take some time? On a slow network we wouldn't want to delay showing the notification, right? Ideally, I guess we would show the notification immediately and update it with the artwork once it is loaded? But that can get complicated. Maybe we can start simpler and wrap the `await()` call on the artwork in [`withTimeout`](https://kotlinlang.org/docs/reference/coroutines/cancellation-and-timeouts.html#timeout) so that we do not wait too long for it? That may be a good compromise for now, I guess?

Reply:
A timeout could be a good idea.  However, showing the notification as soon as we could is not a problem.  Except YouTube, most of the other streaming sites don't have metadata.  Our implementation already put up a notification with the default title and default artwork immediately when a media starts playing.